client/clientimpl_test.go

```diff
@@ -465,6 +465,85 @@ func TestIncludesDetailsOnReconnect(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestStopFromCallback(t *testing.T) {
```

Comment:
Note: before this PR's fix this test hangs in Stop(). This test verifies that the fix works correctly.